Bottom: 3e69dd8b87405c8918da8978e1412c95d681b1a8
Top:    facc0b7417d101a364ef97d4c9bbae5942e7e109
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 12:29:21 +0800

Refresh of autofs-5.0.7-allow-use-of-hosts-map-in-maps.patch

---

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index de09f2b..9a12028 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -1293,15 +1293,6 @@ int parse_mount(struct autofs_point *ap, const char *name,
 		return 1;
 	}
 
-	/*
-	 * Transform hosts map into a map type so we don't confuse the
-	 * options parsing.
-	 */
-	if (strstr(pmapent, "-hosts")) {
-		char *t = strstr(pmapent, "-hosts");
-		*t++ = 'h'; *t++ = 'o'; *t++ = 's'; *t++ = 't'; *t++ = 's';
-		*t = ':';
-	}
 	p = skipspace(pmapent);
 
 	/* Deal with 0 or more options */
@@ -1443,6 +1434,8 @@ int parse_mount(struct autofs_point *ap, const char *name,
 			p += l;
 			p = skipspace(p);
 
+			error(LOGOPT_ANY, "options %s", options);
+
 			l = parse_mapent(p, options, &myoptions, &loc, ap->logopt);
 			if (!l) {
 				cache_delete_offset_list(mc, name);
